components/feature/intent/src/main/java/mozilla/components/feature/intent/IntentProcessor.kt

```diff
@@ -39,71 +36,28 @@ class IntentProcessor(
     private val useDefaultHandlers: Boolean = true,
     private val openNewTab: Boolean = true,
     private val isPrivate: Boolean = false
```

Comment:
Looking at the way Fenix uses [the intent processor](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/Utilities.kt#L28-L34) they create two separate ones for private/regular tabs, so that would mean separate processors for custom tabs/PWAs in private/regular states as well.

For this reason I originally thought it would be nice to separate out the `private` flag to be a separate processor, but I like this direction a lot. I wonder if @pocmo is aware of any pitfalls from having two instances of an IntentProcessor with each having their own CT/PWA processors.

Reply:
Long term I want to update Fenix to use two `TabIntentProcessors` + other processors and entirely get rid of the fat `IntentProcessor`.

Reply:
Oh perfect! That's along the similar lines that I was thinking. ➕ 💯